sway: 1.6.1 -> 1.7

primeos opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

TODOs:
- [x] The terminal emulator in the default config file has been changed to foot.
- [x] Fix the VM test
- [x] Testing
- [x] Squash the commits

Known issues:
- Chromium/Electron: Ozone/Wayland is broken
- When still running Sway 1.6.1 and `swaynag` is from 1.7 it will crash with:
  ```
  [michael@quorra:~]$ swaynag -m test
  wl_registry@2: error 0: invalid version for global wl_output (39): have 3, wanted 4
  00:00:00.000 [swaynag/swaynag.c:438] Error during outputs init.
  ```
- Dropping Alacritty from the NixOS module can require configuration changes

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

primeos committed:
sway: 1.6.1 -> 1.7

Release notes: https://github.com/swaywm/sway/releases/tag/1.7

Notable (backward incompatible) changes:
- The default terminal changed from Alacritty to foot

Known issues:
- `swaynag` will crash when Sway 1.6.1 is still running while the Nix
  package (and thus `swaynag`) is already updated to version 1.7.
- The experimental Ozone/Wayland support of Electron apps will be broken
  for a while. Electron version 17 should work but the Chromium fixes
  haven't yet been backported to Electron version 16.

NixOS module: programs.sway.extraPackages: The "alacritty" package was
replaced with "foot".

VM test: We switched from the OpenGL ES 2.0 renderer to Pixman. The
terminal was also changed to foot but Alacritty is still used for the
XWayland test (since foot doesn't support X11).

Co-authored-by: Patrick Hilhorst <git@hilhorst.be>

Synthetica9 commented:
I'm this close 🤏🏻 to suggesting we just add a `sway_default_terminal`
symlink and patch the config to use that instead of keeping all old
terminals around for ~5 releases

On Thu, 23 Dec 2021, 20:38 ofborg[bot], ***@***.***> wrote:

> @ofborg <https://github.com/ofborg>[bot] requested your review on: #151902
> <https://github.com/NixOS/nixpkgs/pull/151902> sway: 1.6.1 -> 1.7.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/151902#event-5811349555>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABV7PJ25OTSPSDZWFYOUUZTUSN3BXANCNFSM5KVLSJNQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

Synthetica9 commented:
Test also broke, see here:

![image](https://user-images.githubusercontent.com/7075751/147549539-e91e3b6d-30cb-4df2-813c-3e42e01aca17.png)

Synthetica9 commented:
vscod{e,ium} with ozone platform wayland also seem broken, but I assume that's a wlroots thing rather than a Sway thing

primeos commented:
@Synthetica9 thanks for the testing and feedback! :) Unfortunately I'm currently quite behind with testing as I wanted for Hydra to build more of staging-next but it's progressing slowly / will take a while.

> I'm this close 🤏🏻 to suggesting we just add a `sway_default_terminal`
symlink and patch the config to use that instead of keeping all old
terminals around for ~5 releases

That would also be an option. But then again I'm hoping that this'll be the last change of the default terminal emulator for a while. Last time we dropped `rxvt-unicode` quite late but already mentioned that `foot` will become the new default: https://github.com/NixOS/nixpkgs/commit/eb8a694d92332b62769e5b880a93b9ba96883e26

Given that `sway_default_terminal` won't help with copied/adapted configurations I'd suggest we simply make the change right away this time. It isn't ideal but then again every Sway user should have no issue figuring the problem out (it'll likely be a bit annoying and especially unexpected that `$mod+Enter` will likely be broken but usually everyone should use some sort of `dmenu` program to quickly investigate without even having to leave the Sway session; it's quite convenient that most Sway users should be power-users :D).

> Test also broke, see here:

I've added it to the TODOs, thanks. The full (debug) output should hopefully contain enough information to find a fix/workaround.

> vscod{e,ium} with ozone platform wayland also seem broken, but I assume that's a wlroots thing rather than a Sway thing

Oh, that doesn't sound good... :o Hopefully it doesn't affect the current Chromium as well. Unfortunately Ozone/Wayland is still far from stable (I forgot the official status but IIRC Igalia still considers it alpha or beta).

Synthetica9 commented:
Chromium also broke, definitely a blocker then:

```sh
$ chromium --verbose --enable-features=UseOzonePlatform --ozone-platf
orm=wayland
interface 'wl_output' has no event 4
[1229/184110.716325:ERROR:process_memory_range.cc(75)] read out of range
fish: Job 1, 'chromium --verbose --enable-fea…' terminated by signal SIGTRAP (Trace or breakpoint trap)
$ rwhich chromium
/nix/store/ds5pnbnjhdb96azyisgmkikhw0g0c65q-chromium-96.0.4664.110/bin/chromium
```

tadeokondrak commented:
> Chromium also broke, definitely a blocker then:

This is a Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1279574 ([patch](https://source.chromium.org/chromium/chromium/src/+/dd4c3ddadbb9869f59cee201a38e9ca3b9154f4d))

Synthetica9 commented:
> This is a Chromium issue: [bugs.chromium.org/p/chromium/issues/detail?id=1279574](https://bugs.chromium.org/p/chromium/issues/detail?id=1279574) ([patch](https://source.chromium.org/chromium/chromium/src/+/dd4c3ddadbb9869f59cee201a38e9ca3b9154f4d))

Yikes, this might mean Electron apps may stay broken for a while (especially since we don't build most of those)...

primeos commented:
I've "fixed" the VM test by switching to the new Pixman renderer. It's more of a workaround than a fix but we already had to use software rendering before and this should work more reliably for the VM test.

I currently lack time for a more thorough analysis but from a quick test wlroot's TinyWL doesn't seem affected.

> Yikes, this might mean Electron apps may stay broken for a while (especially since we don't build most of those)...

That is indeed really unfortunate... :o
I haven't seen a backport of the fixes yet so they might only land in Chromium 99 (https://github.com/chromium/chromium/commit/dd4c3ddadbb9869f59cee201a38e9ca3b9154f4d and https://github.com/chromium/chromium/commit/a84b79daa8897b822336b8f348ef4daaae07af37). Chromium 99 is scheduled to be released on 2022-03-01 so we should backport the fixes.

primeos commented:
@ofborg test sway

Synthetica9 commented:
Updated the test to use foot.

> That is indeed really unfortunate... :o
> [...]

Would also be possible to fix this from wlroots perhaps? Honestly, I wouldn't feel comfortable upgrading (the default version of) Sway to 1.7 if something I think quite a lot of users use would knowingly break without a good fix... I know I would be staying on Sway 1.6 (for now) if the full version of 1.7 were to release today

jackinloadup commented:
Sounds perfect for unstable. I'd really like to use it personally due to some of the fixes. It doesn't sound like I'll be affected by any of the unfortunate chromium issues. Let me see if I can figure out how to apply patches :wink:  Gotta get DRM leasing for VR headsets. :crossed_fingers:

Synthetica9 commented:
> Sounds perfect for unstable.

While nixos unstable _is_ a rolling-release distro, in my experience it is also one of the most stable rolling-release distros out there. I'd like to avoid knowingly breaking things if I can avoid it.

>  It doesn't sound like I'll be affected by any of the unfortunate chromium issues. [...] Gotta get DRM leasing for VR headsets. 

Haha yeah I guess I'm biased towards the chromium issue because I use a lot of electron-based apps and have a mixed-scaling setup, which isn't great with XWayland so I use as many wayland-native apps as I can get away with :P

> Let me see if I can figure out how to apply patches 😉 

The problem is that we don't build a lot of the electron apps ourselves, but grab pre-built binaries for them, so it would be difficult to patch those.

primeos commented:
@Synthetica9 thanks for https://github.com/NixOS/nixpkgs/pull/151902/commits/0860d0db4f74770677011bf9c95a1d76b38ba512 but can you please drop https://github.com/NixOS/nixpkgs/pull/151902/commits/5ec09505462c098e8189b611932a8a3795c978a8? It just makes working on this PR unnecessarily annoying (I intend to squash all commits at the end btw - assuming it doesn't really help to keep some changes separate - therefore this PR should also only contain the relevant changes for the update) and doesn't belong in this PR ("destroys" the diff; feel free to reformat it after that).

> I'd like to avoid knowingly breaking things if I can avoid it.

+1, the name nixos-unstable is misleading, unfortunately. However, in this case I'm not sure yet how to proceed. One possibility would be to add `sway_1_6` and `sway_1_7` and start with `sway = sway_1_6` if necessary. However, the NixOS module intentionally doesn't offer a `package` option so this is still not that convenient (requiring e.g. an overlay).

In any case the goal should be to (eventually) update Sway to version 1.7 by default.

> The problem is that we don't build a lot of the electron apps ourselves, but grab pre-built binaries for them, so it would be difficult to patch those.

Agreed, the current status sucks for Electron users... :o

> Would also be possible to fix this from wlroots perhaps?

I don't think they could apply a good workaround (without downgrading the supported `wl_outputs` version for all applications). I haven't looked at the details though. But given that they're aware of it and the wlroots bug report got closed (https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3344#note_1189198) I don't think there's any interest in a workaround (understandably).

I think the better option would be to try to get the Chromium patches backported to older releases and Electron (ideally that would already be happening as this should affect other Wayland compositors soon as well, I guess).

Synthetica9 commented:
> @Synthetica9 thanks for [0860d0d](https://github.com/NixOS/nixpkgs/commit/0860d0db4f74770677011bf9c95a1d76b38ba512) but can you please drop [5ec0950](https://github.com/NixOS/nixpkgs/commit/5ec09505462c098e8189b611932a8a3795c978a8)? 

✅

> One possibility would be to add sway_1_6 and sway_1_7 and start with sway = sway_1_6 if necessary.

Might honestly be the best option in this case, re-evaluate once all or most electron packages have the relevant patch.

>  However, the NixOS module **intentionally doesn't offer a package option** so this is still not that convenient (requiring e.g. an overlay). 
> 
> (emphasis mine)

What's the reasoning behind this? 

> I don't think there's any interest in a workaround (understandably).

Yeah, I fully understand the sentiment of the wlroots people to not want to do such a workaround, but but it leaves us in a difficult situation downstream.

primeos commented on nixos/tests/sway.nix:
```diff
@@ -66,30 +80,28 @@ import ./make-test-python.nix ({ pkgs, lib, ...} :
-    print(machine.succeed("cat /tmp/test-x11.out"))
-    machine.screenshot("alacritty_glinfo")
+    machine.copy_from_vm("/tmp/test-x11.out")
+    machine.screenshot("alacrittyglinfo")
```
Was the removal of the underscore intentional (below you're using `foot_wayland_info`)?

Synthetica9 replied:
no, I was running into another bug (#153766), thought it had something to do with the underscore and never changed it back.

primeos commented on nixos/tests/sway.nix:
```diff
@@ -66,30 +80,28 @@ import ./make-test-python.nix ({ pkgs, lib, ...} :
     machine.wait_for_file("/run/user/1000/wayland-1")
     machine.wait_for_file("/tmp/sway-ipc.sock")
 
-    # Test XWayland:
+    # Test XWayland (foot does not support X):
     machine.succeed(
         "su - alice -c 'swaymsg exec WINIT_UNIX_BACKEND=x11 WAYLAND_DISPLAY=invalid alacritty'"
     )
     machine.wait_for_text("alice@machine")
     machine.send_chars("test-x11\n")
     machine.wait_for_file("/tmp/test-x11-exit-ok")
-    print(machine.succeed("cat /tmp/test-x11.out"))
-    machine.screenshot("alacritty_glinfo")
+    machine.copy_from_vm("/tmp/test-x11.out")
```
The `copy_from_vm` is very cool, thanks!
I'm just wondering if we should keep the `print`. My main reason for that was that it's available via the build logs as well (ofborg and Hydra).

Synthetica9 replied:
We could have both if we want

primeos commented on nixos/tests/sway.nix:
```diff
@@ -66,30 +80,28 @@ import ./make-test-python.nix ({ pkgs, lib, ...} :
-    machine.succeed(
-        "su - alice -c 'swaymsg exec WINIT_UNIX_BACKEND=wayland DISPLAY=invalid alacritty'"
-    )
+    machine.sleep(3)
```
Out of curiosity: Is that sleep required when switching away from `swaymsg` or is this just out of precaution?

Synthetica9 replied:
I was running into issues of the terminal not opening when I didn't have it. Maybe it could be shorter

primeos replied:
Ok, thanks for all of your feedback. Should be fine, I was just curious.

primeos reviewed (commented):
> What's the reasoning behind this? (package option)

https://github.com/NixOS/nixpkgs/issues/50476 (it wasn't necessary so far) but it seems like I forgot about the last update: https://github.com/NixOS/nixpkgs/issues/50476#issuecomment-637841528

So I guess it could make sense to add such an option now (too choose between Sway 1.6 and 1.7 - apart from that there's probably little use for it as overlays can already be used as an alternative and AFAIK there are no relevant Sway alternatives/forks).

> Yeah, I fully understand the sentiment of the wlroots people to not want to do such a workaround, but but it leaves us in a difficult situation downstream.

Yes, unfortunately. Regarding Chromium: I tested the fixes with M98 (https://github.com/NixOS/nixpkgs/pull/153633) and it worked but I got a few other errors that broke hardware acceleration (but I need to check if that's a regression or not as I haven't used Chromium on that system for quite a while...).

LunNova commented:
Might be worth trying with `WLR_RENDERER=vulkan` `VK_LAYER_PATH = "${pkgs.vulkan-validation-layers}/share/vulkan/explicit_layer.d"` instead of pixman?

Synthetica9 commented:
> Might be worth trying with `WLR_RENDERER=vulkan` `VK_LAYER_PATH = "${pkgs.vulkan-validation-layers}/share/vulkan/explicit_layer.d"` instead of pixman?

Crashes with a similar error:

```
<<< Welcome to NixOS 22.05pre-git (x86_64) - tty2 >>>


machine login: alice (automatic login)


[alice@machine:~]$ sway
00:00:00.126 [wlr] [render/vulkan/renderer.c:1487] Could not match drm and vulkan device
00:00:00.126 [sway/server.c:79] Failed to create renderer

[alice@machine:~]$
```

primeos commented:
I think Pixman should be fine for now. We needed `WLR_RENDERER_ALLOW_SOFTWARE` before so that wasn't ideal either.

For other renderers we should look into getting proper hardware acceleration to work:

https://github.com/NixOS/nixpkgs/blob/02a162f479aa17a90212acc613c1117a25170d0e/nixos/tests/sway.nix#L20-L23

I wouldn't consider it a high-priority TODO but feel free to have a look at it if someone has time and motivation.

IIRC it might at least require changes to Mesa as well since we currently don't even build VirGL by default (I enabled it once but it still wasn't enough - maybe that's changed though). For Vulkan there's also Venus now: https://docs.mesa3d.org/drivers/venus.html
Another problem might've been that 3D acceleration with QEMU requires the `-sdl` (`-display sdl`) flag while our VM tests run headless. Anyway, not sure how the current situation looks but I definitely had 3D hardware-acceleration working with QEMU/KVM on NixOS (but that wasn't a headless setup).

FlorianFranzen commented:
Updating to a newer wlroots' version will also mean we will need [this patch](https://chromium.googlesource.com/chromium/src/+/dd4c3ddadbb9869f59cee201a38e9ca3b9154f4d) in all chromium based applications.

primeos commented:
Yes, we already noticed and discussed it here (s. above). Chromium in Nixpkgs is also patched since today (https://github.com/NixOS/nixpkgs/pull/155203).

> +1, the name nixos-unstable is misleading, unfortunately. However, in this case I'm not sure yet how to proceed. One possibility would be to add sway_1_6 and sway_1_7 and start with sway = sway_1_6 if necessary.

In the meantime I thought a bit more about this. I'm now for defaulting Sway to version 1.7. This isn't a Sway/wlroots bug, Chromium is now patched, and Electron apps would have to be used via XWayland in the meantime (IMO this is fair since Ozone/Wayland is still in alpha (or beta?)). It's certainly not ideal but IMO by far not enough to hold the update back.

If deemed necessary we could add a `programs.sway.package` option. Any opinions on this? Or maybe someone has a better idea?

Synthetica9 commented:
> Updating to a newer wlroots' version will also mean we will need [this patch](https://chromium.googlesource.com/chromium/src/+/dd4c3ddadbb9869f59cee201a38e9ca3b9154f4d) in all chromium based applications. Otherwise, due to the new protocol version, all these apps will crash with `interface 'wl_output' has no event 4`.

This was also discussed above, starting at https://github.com/NixOS/nixpkgs/pull/151902#issuecomment-1002157431

> The patch was included in `v99.0.4766`, so it already hit electron master (v18 nightly, with [request for backport](https://github.com/electron/electron/issues/32487)) and chromium dev (#154668).

Thank you for keeping an eye on this, I hope this will trickle down faster than we expect!

Synthetica9 commented:
@ofborg test sway

LunNova commented:
For apps which bundle their own outdated electron versions, it's usually possible to use an alternative electron version.

Could consider doing this for apps which stop working with this update.

Example arch build doing this for discord https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=discord_arch_electron

chvp commented:
1.7 final was released yesterday

primeos commented:
Ok, so from my point of view this is ready to be merged as is. I'm running RC3 on my main setup since it was released and didn't notice any new issues so far (apart from not being able to use the Wayland support of Electron apps (but luckily I don't need any Electron apps anymore) - that isn't a Sway bug though and there's still XWayland or nested Wayland compositors). The diff also looks good to me (a few additional changes but still a clean diff). So I suggest we merge this :) Or are there any objections left?

Synthetica9 commented:
Do we want to keep 1.6 around for the time being to accommodate Electron
users?

On Sun, 23 Jan 2022, 19:46 Michael Weiss, ***@***.***> wrote:

> Ok, so from my point of view this is ready to be merged as is. I'm running
> RC3 on my main setup since it was released and didn't notice any new issues
> so far (apart from not being able to use the Wayland support of Electron
> apps (but luckily I don't need any Electron apps anymore) - that isn't a
> Sway bug though and there's still XWayland or nested Wayland compositors).
> The diff also looks good to me (a few additional changes but still a clean
> diff). So I suggest we merge this :) Or are there any objections left?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/151902#issuecomment-1019543649>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABV7PJZYWS3J3N2XD5QJOY3UXREJHANCNFSM5KVLSJNQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

primeos commented:
> Do we want to keep 1.6 around for the time being to accommodate Electron
users?

That's the question. I've written about it in https://github.com/NixOS/nixpkgs/pull/151902#issuecomment-1013887127 but there were not replies. I don't feel that it's necessary (we already provide override mechanics) and if we'd add one we'd also need `programs.sway.package` to make it usable without an overlay (and if users could figure out that they need to downgrade Sway to fix Electron they could very likely also do so via an overlay). And the biggest reason against it is that apparently the Wayland support of Electron 16 is also broken since 16.0.6 (or 16.0.5). At least I noticed that with Signal-Desktop and other Linux distributions are affected as well: https://github.com/NixOS/nixpkgs/pull/154003/commits/e06082eda06ee586de17e0a060e8cd0eddd20c69. That should theoretically affect more Electron apps soon.

Anyway, I'm not objecting if someone thinks that having `sway_1_6` would help.

dmayle commented:
Personally, I'm waiting for this to land.  Electron works under XWayland, and there are other override mechanisms, so the only argument in favor of keeping sway_1_6 is to enable users to choose a third workaround mechanism as opposed to one of two others.  Given the added maintenance cost, I can't find the value.

Synthetica9 commented:
Okay, feel free to merge then. I'm opening a tracking issue for the Chromium situation

Princemachiavelli commented:
> Personally, I'm waiting for this to land. Electron works under XWayland, and there are other override mechanisms, so the only argument in favor of keeping sway_1_6 is to enable users to choose a third workaround mechanism as opposed to one of two others. Given the added maintenance cost, I can't find the value.

Agreed, my impression is having multiple explicit versions of packages really should be limited to packages with explicit versions that are often pinned such as programming languages/runtimes and databases i.e it's a configuration supported by upstream. If someone really needs to hold back a package completely there are methods for using overlays in their Nix config. I would rather see those methods become better documented and used more frequently than for the nixpkgs repository to accumulate more and more workarounds.

primeos commented:
Ok, thanks for the quick feedback. Let's merge this then :) :rocket:

I also want to thank everyone who helped via reviews/feedback, etc. (i.e. all participants of this PR).

Synthetica9 commented:
![pre](https://user-images.githubusercontent.com/7075751/150699386-11a8a631-ce95-4ea3-aae0-3138186e327a.png)

Weird glitch I encountered in testing, and I have no idea if it's something that could pop up in the future. Just wanted to share it here, but assuming it was a weird one-of, can't reproduce it now.

Princemachiavelli commented:
@Synthetica9 What GPU/GPU driver are you using? I'm an Vega 8/amdgpu and haven't had any regressions. Although `sway --version` does show "sway version 1.8-dev" instead of 1.7 . probably an issue with the release commit.

I've been using chromium and several Electron apps like VScode, Electronmail, and Freetube and they all seem to work so they must all be defaulting to XWayland.

The only thing I have noticed is that the new `render_bit_depth` option does not result in the GPU/display actually switching to a 10 bit profile; drm_info still shows the current  bpc as 8. Even with the verbose and debug flags I don't see any errors related to switching the render/display profile so I'm not sure what the issue is.

Synthetica9 commented:
> @Synthetica9 What GPU/GPU driver are you using? I'm an Vega 8/amdgpu and haven't had any regressions. Although sway --version does show "sway version 1.8-dev" instead of 1.7 . probably an issue with the release commit.

This is in the QEMU test driver, on an Intel GPU.
